### DISCUSSION TITLE
How to use CEBRA for epoched data?

### DISCUSSION THREAD

**Discussion by timonmerk** [AdaptiveMotorControlLab/CEBRA#49] [Q&A] [answered]
Thanks again for being so responsive in the discussion here!

We are wondering how you would recommend using CEBRA for epoched or trial-wise data. The paper Figure 3 is based on trial specific data, and it seems that the trials were concatenated s.t. `monkey_target.neural` is of shape number of neurons x time (https://cebra.ai/docs/demo_notebooks/Demo_primate_reaching.html).

There are a couple of options to deal with epochs. First, they can be concatenated, but this would mean that behaviorally distant samples are fed in the same "sample" to the network, the receptive field would include overlapping information from two epochs. This might be problematic if epochs are short and the receptive fields need to be large.

Next, the model could be fitted multiple times with the  `partial_fit` method. But that would make adapting the learning challenging, since the order of epochs during training would have a significant effect on final weight distribution.

So I am wondering how you would train the model with this kind of data?

**Comment by MMathisLab** [comment, marked as answer]
No problem! Indeed in our first paper we used more or less continuous data (and in Fig. 3 that was how the others in benchmarks used it). If you don't have access to continuous data then concatenating might not be ideal (as you would sample over junctions that likely don't make sense). So, two options: (1) use trial ID as a auxiliary variable, or (2) wait until we release a new update with this (which is already in the works 🤗), but I don't have a firm date on this update yet.

**Reply by timonmerk** [reply]
Thank you @MMathisLab!

**Reply by psulewski** [reply]
Dear @MMathisLab, is there any news on the update regarding the epoch/trial based data yet? 
That would be super useful on our end as well.

Thank you very much,
Best Philip

**Reply by xanderladd** [reply]
+1 -- following for updates
